net/FlatBuffers/ByteBuffer.cs

```diff
@@ -63,11 +65,65 @@ public void Reset()
             _pos = 0;
         }
 
+        // Create a new ByteBuffer on the same underlying data.
+        // The new ByteBuffer's position will be 0, but starting
+        // from this ByteBuffer's current position.
+        public virtual ByteBuffer Slice()
+        {
+            return new ByteBufferSlice(_buffer, 0, Position);
+        }
+
+        // Increases the size of the ByteBuffer, and copies the old data towards
+        // the end of the new buffer.
+        public virtual void GrowFront(int newSize)
```

Comment:
Where did this function come from / where is it called from?

Reply:
By hiding `Data`, I moved growing the buffer from `FlatBufferBuilder` to `ByteBuffer`. This is a 1:1 copy of that method. It's called from `FlatBufferBuilder::GrowBuffer`.